src/options.ts

```diff
@@ -94,12 +99,20 @@ export function _setDefaultOptions(options: Partial<Options> = {}): Options {
   const serviceName =
     options.serviceName ||
     process.env.OTEL_SERVICE_NAME ||
-    resource.attributes[SemanticResourceAttributes.SERVICE_NAME] ||
-    defaultServiceName;
+    resource.attributes[SemanticResourceAttributes.SERVICE_NAME];
+
+  if (!serviceName) {
```

Comment:
Is `diag.warn` visible in the terminal as well?

Reply:
I'll merge it, @seemk, but any comments are still welcome!